### PR TITLE
CR-1144981 hwmon_sdm: fix memcpy() overflow issue

### PR DESCRIPTION
**Pull request by rajkumar-xilinx**
Signed-off-by: Rajkumar Rampelli <rampelli@amd.com>

#### Problem solved by the commit
memcpy overflow error seen when sensor data UUID received from VMR has more size (17 bytes) than hwmon_sdm driver's structure member UUID (8 bytes) size.
#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
CR-1144981
#### How problem was solved, alternative solutions (if any) and why they were rejected
Set uuid member size to BDINFO data length which is 32 bytes by default.
#### Risks (if any) associated the changes in the commit
NA
#### What has been tested and how, request additional testing if necessary
Read UUID data through hwmon_sdm sysfs node.
Examine reports using xbutil & xbmgmt tool.
#### Documentation impact (if any)
NA

### CODE REVIEW

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c [hunk 1]
```diff
@@ -1307,8 +1307,14 @@ static ssize_t
 uuid_show(struct device *dev, struct device_attribute *attr, char *buf)
 {
 	struct xocl_hwmon_sdm *sdm = dev_get_drvdata(dev);
+	int count = 0, i = 0;
 
-	return sprintf(buf, "0x%llx\n", sdm->bdinfo.uuid);
+	count += sprintf(buf, "0x");
+	for (i = 0; i < strlen(sdm->bdinfo.uuid); i++)
```
Comment (houlz0507) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c:1313]: Using strlen() does not look right. What if there is 0 in uuid[] ?

Comment (keryell) [reply src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c:1313]: It makes also the loop complexity O(n2) because strlen is already O(n)...

Comment (houlz0507) [reply src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c:1313]: @rajkumar-xilinx I assume big endian in bdinfo.uuid. you can use %pUb (lower case) or %pUB(upper case). The output buffer format is "aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee" and the length should be UUID_STRING_LEN + 1.

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c [hunk 2]
```diff
@@ -70,7 +70,7 @@ struct xocl_sdr_bdinfo {
 	char revision[SDR_BDINFO_ENTRY_LEN_MAX];
 	uint64_t mfg_date;
 	uint64_t pcie_info;
-	uint64_t uuid;
+	char uuid[SDR_BDINFO_ENTRY_LEN];
```
Comment (rajkumar-xilinx) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c:73]: 16 bytes

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c (hunk 1 above)
Comment (rajkumar-xilinx) [reply src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c]: Thanks @houlz0507 your suggestion worked well.

**Comment by rajkumar-xilinx** [comment]
$ cat /sys/bus/pci/devices/0000\:3b\:00.0/hwmon_sdm.m.56623104/uuid
1775868F-57CE-8BA4-D549-3643C36C895B

**Review by chvamshi-xilinx — APPROVED** [review]

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c [hunk 3]
```diff
@@ -70,7 +70,7 @@ struct xocl_sdr_bdinfo {
 	char revision[SDR_BDINFO_ENTRY_LEN_MAX];
 	uint64_t mfg_date;
 	uint64_t pcie_info;
-	uint64_t uuid;
+	char uuid[UUID_STRING_LEN];
```
Comment (rajkumar-xilinx) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c:73]: + 1

**Review by houlz0507 — APPROVED** [review]

**Comment by rajkumar-xilinx** [comment]
retest this please

**Review by maxzhen — APPROVED** [review]

**Comment by chvamshi-xilinx** [comment]
Merging as this is required for new platform

**Comment by chvamshi-xilinx** [comment]
@rajkumar-xilinx , please backport it to 2022.2

**Pull request merged**